messaging/WFLY-10493_support_java_ee_resource_definitions_remote_artemis.adoc

```diff
@@ -0,0 +1,168 @@
+Using the properties on a DestinationDefinition we can pass a `management-address` property to define the management address to use for sending the management messages to the remote Artemis broker.
+
+== Proposed Design
+
+This feature requires no change to the `messaging-activemq` subsystem structure.
+
+We will be using the `resourceAdapter` parameter to look for pooled-connection-factories either local or remote.
+
+The Java EE7 specification also expects that a JMS resource definitions works out of the box without having to specify a `resourceAdapter` name.
+
+The new behaviour will be:
+For destinations:
+----
+1. if `resourceAdapter` attribute is specified or one can be defined from the default-binding for the `jms-connection-factory`
+  1.1 if a `pooled-connection-factory` with the specified name exists
+        1.1.1 if it is a `pooled-connection-factory` to a remote broker: use it to connect to the remote broker and send management messages.
+        1.1.2 else it is a 'local' broker : use the `server` resource to create the required destinations.
+2. else
+    2.1 delegate to the `resource-adapters` subsystem (and IronJacamar) to create the JMS resources based on the resource adapter's `admin-objects` and `connection-definitions` resources.
+----
+
+=== Examples
+
+==== Use the default resource adapter
+
+It corresponds to [1.1.1] of the new behavior.
+
+* create a connector `/subsystem=messaging/remote-connector=remote-amq(socket-binding="messaging-remote-throughput")`
+* create a `/subsystem=messaging/pooled-connection-factory=activemq-ra-remote(entries=["java:/jms/remote-amq/JmsConnectionFactory"], connectors=["remote-amq"])` resource
+* define `/subsystem=ee/service=default-bindings:write-attribute(name=jms-connection-factory="java:/jms/remote-amq/JmsConnectionFactory")`
+* use the annotation
+
+[source,java]
+----
+@JMSConnectionFactoryDefinition(
+        name="java:app/myCF"
+)
+----
+
+==== Use a remote Artemis broker
+
+It corresponds to [1.1.1] of the new behaviour.
+
+* use the annotation
+
+[source,java]
+----
+@JMSConnectionFactoryDefinition(
+        name="java:app/myCF"
+        resourceAdapter="myPCF"
+)
+----
+
+==== Use a 3rd party JMS resource adapter
+
+It corresponds to [2.1] of the new behaviour.
+
+* define a `resource-adapter` (e.g. named `wsmq`) in the `/subsystem=resource-adapters` subsystem.
+* use the annotation
+
+[source,java]
+----
+@JMSConnectionFactoryDefinition(
+        name="java:app/myCF"
+        resourceAdapter="wsmq"
+)
+----
+
+==== Use a remote Artemis broker with a specific management.queue
+
+It corresponds to [1.1.1] of the new behaviour.
+
+* use the annotation
+
+[source,java]
+----
+@JMSDestinationDefinition(
+    name = "java:/jms/queue/MessageBeanQueue",
+    interfaceName = "javax.jms.Queue",
+    destinationName = "MessageBeanQueue"
+    properties= {
+        "management-address=remote-activemq.management"
```

Comment:
I have concern with that property and its implication on the feature support.
This is an implementation detail (how WildFly can ask the remote Artemis to create the resources corresponding to the JMS destination) that is leaking to the user codebase.

If Artemis was supporting JCA AdminObject in its resource adapter, we would not need a specific way to configure annotations for a remote Artemis broker.
The messaging-activemq subsystem already supports configuring JMS destinations with 3rd-party resource adapter using AdminObject.

I'd prefer we open a RFE with Artemis so they support Admin Object for JMS Qeueus and Topics.
Their RA has all the information to connect to the remote servers and this would  avoid a lot of code in the subsystem to "extract" these informations for Artemis RA (and the `pooled-connection-factory` resource).